compiler-core/src/type_/hydrator.rs

```diff
@@ -183,7 +184,19 @@ impl Hydrator {
                 }
 
                 // Ensure that the correct number of arguments have been given to the constructor
-                if arguments.len() != parameters.len() {
+                //
+                // This is a special case for when the type expects no parameters
```

Comment:
Either a type has parameters, or a type constructor expects arguments. A type does not expect arguments.